pkg/server/api/agent/v1/service.go

```diff
@@ -33,6 +41,8 @@ func RegisterService(s *grpc.Server, service *Service) {
 
 // Config is the service configuration
 type Config struct {
+	Catalog     catalog.Catalog
+	Clock       clock.Clock
```

Comment:
When calling the `New` function, `Clock` needs to be initialised (https://github.com/spiffe/spire/blob/master/pkg/server/endpoints/config.go#L130)